R/duplicates.R

```diff
@@ -185,3 +185,62 @@ throw_msg_if_duplicates <- function(dt, assay_name = "unknown", msg_f = stop, pr
      msg_f(paste0(msg, msg2, msg3))
   }
 }
+
+#' send email if assay data.table contains duplicated rows
+#' 
+#' An auxiliary function to send email if duplicated rows in assay data.table are found
+#' 
+#' @param dt data.table with assay data
+#' @param by charvec with notification methods (currently 'email' supported)
+#' @param assay_name string with the name of the assay
+#' @param preview_max_numb number of rows to preview if duplicates found
+#' @param metadata list with the additional metadata to send via email
+#' @examples
+#' sdata <- get_synthetic_data("finalMAE_small")
+#' smetrics_data <- convert_se_assay_to_dt(sdata[[1]], "Metrics")
+#' throw_msg_if_duplicates(smetrics_data, assay_name = "Metrics", msg_f = futile.logger::flog.info)
+#' @return NULL
+#' @keywords duplicates
+#'
+#' @export
+#' 
+notify_if_duplicates <- function(dt, by = "email", assay_name = "unknown", preview_max_numb = 4, metadata = NULL) {
```

Comment:
Wasn't be that nor simple to move `notify_if_duplicates` to gDRcomponents? As well as `send_email`?

Thx to that:
- `get_env_var` could stay also in gDRcomponets with its family of setters and getters 
- gDRutils would stay a package that operates on the gDR objects and datasets 
- gDRcomponets - package that supports gDR viz app.

Reply:
Ok. I will move `notify_if_duplicates` and `send_email` to gDRcomponents. 
I will keep `get_env_var` in gDRutils. It can be used across all gdrplatform packages.

Reply:
done.

Reply:
✔️